src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -380,6 +400,25 @@ async def common_validate(self, new, old, verrors):
                 "AD domain name is required."
             )
 
+    @accepts()
+    async def config(self):
+        ha_mode = SMBHAMODE[(await self.middleware.call('smb.get_smb_ha_mode'))]
+        if ha_mode != SMBHAMODE.CLUSTERED:
+            return await super().config()
+
+        rv = await self.middleware.call('directoryservices.get_conf',
+                                        self._config.datastore)
+        globals = (await self.middleware.call('smb.reg_globals'))['raw']
+        rv.update({
+            'domainname': globals.get('realm', ''),
+            'use_default_domain': globals.get('winbind use default domain', 'No') == 'Yes',
+            'allow_trusted_domains': globals.get('allow trusted domains', 'No') == 'Yes',
+            'ads_dns_updates': globals.get('ads dns updates', 'No') == 'Yes',
+            'nss_info': globals.get('winbind nss info', ''),
+            'enabled': globals.get('security', 'user') == 'ADS',
```

Comment:
`enabled` needs to be `enable` or a `KeyError` will be raised on line 1689 as well as the `def config()` method in cache.py